include/openthread/platform/dns.h

```diff
@@ -56,6 +56,23 @@ extern "C" {
  */
 typedef struct otPlatDnsUpstreamQuery otPlatDnsUpstreamQuery;
 
+/**
+ * Indicates whether upstream DNS query functionality is available on the platform.
+ *
+ * This function allows the platform to inform the OpenThread stack if, for example, no upstream DNS server is
```

Comment:
```suggestion
 * This function allows the platform to inform the OpenThread stack if no upstream DNS server is
```

Sounds like this is all it does so we don't need "for example" here